### PR TITLE
Changes to LogHandler to fix /log endpoint

### PR DESCRIPTION
**Pull request by CuriousVini**
Changes to LogHandler to fix /log endpoint

Issue: https://issues.cask.co/browse/CDAP-5508
Build: http://builds.cask.co/browse/CDAP-DUT3930-11

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogHandler.java [hunk 1]
```diff
@@ -103,7 +103,22 @@ private void doGetLogs(HttpResponder responder, LoggingContext loggingContext,
...
+      if (runRecord == null) {
+        readFromTimeMillis = currentTime - TimeUnit.HOURS.toMillis(1);
+      } else {
+        readFromTimeMillis =  TimeUnit.SECONDS.toMillis(runRecord.getStartTs());
+        if (runRecord.getStopTs() != null) {
+          readToTimeMillis = TimeUnit.SECONDS.toMillis(runRecord.getStopTs() + 1);
+        }
+      }
+      
+      TimeRange timeRange = parseTime(fromTimeSecsParam, toTimeSecsParam, readFromTimeMillis, readToTimeMillis,
```
Comment (sagarkapare) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogHandler.java]: seems inconsistent to pass two parameters in seconds and two in milliseconds.

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogHandler.java [hunk 2]
```diff
@@ -103,7 +103,22 @@ private void doGetLogs(HttpResponder responder, LoggingContext loggingContext,
...
+
+      // If runId not specified, fetch past 1 hour logs
+      if (runRecord == null) {
+        readFromTimeMillis = currentTime - TimeUnit.HOURS.toMillis(1);
+      } else {
+        readFromTimeMillis =  TimeUnit.SECONDS.toMillis(runRecord.getStartTs());
+        if (runRecord.getStopTs() != null) {
+          readToTimeMillis = TimeUnit.SECONDS.toMillis(runRecord.getStopTs() + 1);
+        }
+      }
```
Comment (sagarkapare) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogHandler.java]: Actually `adjustRange` method below seems to be already taking care of this, right?

Comment (CuriousVini) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogHandler.java]: Yes. But if runRecord is null, `TimeUnit.SECONDS.toMillis(runRecord.getStopTs() + 1)` will fail.

**Comment by CuriousVini** [comment]
@sagarkapare Thanks for the comments. Comments have been addressed. Please review, when you get a chance.

**Comment by sagarkapare** [comment]
@CuriousVini I think the fix can be simplified.

So the behavior we want is like following when the run level logs are requested:
1. if start and stop is not specified, return all logs for that run.
2. if start or stop is specified explicitly use that while returning the logs.

So changes can be restricted to the `adjustReadRange` method which can take two additional boolean parameters to indicate if start and stop is explicitly specified. 

If start is not specified `fromTimeMillis` can always be set to `runRecord.getStartTs()`
Similarly if stop is not specified `toTimeMillis` can always be set to `runRecord.getStopTs()`

**Comment by CuriousVini** [comment]
Closing this PR and created new https://github.com/caskdata/cdap/pull/5632

**Pull request closed without merging**